packages/@aws-cdk/aws-s3-deployment/test/bucket-deployment.test.ts

```diff
@@ -203,6 +203,21 @@ test('deploy from a local .zip file', () => {
 
 });
 
+test('deploy from a local .zip file when efs is enabled', () => {
+  // GIVEN
+  const stack = new cdk.Stack();
+  const bucket = new s3.Bucket(stack, 'Dest');
+
+  // WHEN
+  new s3deploy.BucketDeployment(stack, 'Deploy', {
+    sources: [s3deploy.Source.asset(path.join(__dirname, 'my-website.zip'))],
+    destinationBucket: bucket,
+    enableTempEfsStorage: true,
+    vpc: new ec2.Vpc(stack, 'Vpc'),
+  });
+
```

Comment:
Add as assertion on the resulting template using `expect(stack).toHaveResource`. 

See https://github.com/aws/aws-cdk/blob/7bc6c6eb14ee73c490caa649aeff509e34eb2c52/packages/%40aws-cdk/aws-s3-deployment/test/bucket-deployment.test.ts#L28-L38

Reply:
Added assertion.